spyder_env_manager/spyder/widgets/main_widget.py

```diff
@@ -292,61 +352,304 @@ def on_close(self):
             self.env_manager_action_thread
             and self.env_manager_action_thread.isRunning()
         ):
-            self.env_manager_action_thread.quit()
+            self.env_manager_action_thread.terminate()
             self.env_manager_action_thread.wait()
 
     # ---- Private API
     # ------------------------------------------------------------------------
 
     def _create_info_environment_page(self, title, message):
-        """Create html page to show while the kernel is starting"""
-        with open(ENVIRONMENT_MESSAGE) as templete_message:
-            environment_message_template = Template(templete_message.read())
+        """
+        Create html page to show if no environment exists to describe basic plugin functionality.
+
+        Parameters
+        ----------
+        title : str
+            Title that the page should show.
+        message : str
+            Content that the page should show.
+
+        Returns
+        -------
+        page : str
+            str representation of the html of the usage instructions page.
+
+        """
+        with open(ENVIRONMENT_MESSAGE) as template_message:
+            environment_message_template = Template(template_message.read())
             page = environment_message_template.substitute(
                 css_path=self.css_path, title=title, message=message
             )
         return page
 
+    def _package_table_context_menu_actions(self, action, package_info):
+        """
+        Handle packages table widget context menu actions.
+
+        The possible actions to choose affect a specific package.
+
+        Parameters
+        ----------
+        action : str
+            The Python environment action to be done.
+            The action should defined on the `spyder_env_manager.spyder.widgets.packages_table.EnvironmentPackagesActions` class.
```

Comment:
```suggestion
            The action should defined on the `EnvironmentPackagesActions` enum class.
```

This line is too long, so I propose to shorten it like this. If you have a better suggestion, please disregard mine.